### PR TITLE
DOCS-1380: Support more platforms

### PR DESCRIPTION
**Pull request by andf-viam**
Add newly supported platforms to `--platform` support table for `viam module upload` command.

Note: Epic is named suggesting that multiple archs could be uploaded at the same time, but I wasn't able to do so with v0.13.0. 

EDIT: Thanks for helpful comments Zack, I've fixed up my misconceptions in the copy!

### CODE REVIEW

File: docs/manage/CLI.md [hunk 1]
```diff
@@ -533,10 +533,22 @@ You may use either argument for the `viam module create` command, but must use `
 
 The `--platform` argument accepts one of the following architectures:
 
-- `darwin/arm64` - macOS computers running the `arm64` architecture, such as Apple Silicon.
-- `darwin/amd64` - macOS computers running the Intel `x86_64` architecture.
-- `linux/arm64` - Linux computers or {{< glossary_tooltip term_id="board" text="boards" >}} running the `arm64` (`aarch64`) architecture, such as the Raspberry Pi.
-- `linux/amd64` - Linux computers or {{< glossary_tooltip term_id="board" text="boards" >}} running the Intel `x86_64` architecture.
+<!-- prettier-ignore -->
+|  Architecture  | Description | Common use case |
+| -------------- | ----------- | --------------- |
```
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/manage/CLI.md:538]: Thank you, I was very confused by missing entries on the CLI help output, and seeming to pick specific archs for `any`s. This is an easy clarify in the docs, thanks so much for pointing this out!

**Comment by andf-viam** [comment]
@zaporter-work Thanks so much! I've updated with your feedback 👍 My apologies on the `any` = "multiple platforms" concept, my brain was stuck expecting something like `--platform any,linux/any,linux/arm64`. It makes perfect sense now that individual `upload` must be per-arch (how else would I adjust my `requirements.txt` or specific build flags per arch). Thanks for helpful clarification!

I chose explicitness on the updated table, with each row getting a definitive use case per your comments. This is likely too much for CLI help text, but I might suggest presenting a truncated version of each row in the help text output: my confusion reading that output was empty desc to some platform rows.

Let me know if you'd like to see any changes!

File: docs/manage/CLI.md [hunk 2]
```diff
@@ -533,15 +533,30 @@ You may use either argument for the `viam module create` command, but must use `
...
+| `darwin/amd64` | macOS machines running the Intel `x86_64` architecture. | Suitable for most C++ or Go modules on macOS `amd64`. |
+| `darwin/arm64` | macOS machines running the `arm64` architecture, such as Apple Silicon. | Suitable for most C++ or Go modules on macOS `arm64`. |
+
+You can use the `uname -m` command on your computer or board to determine its system architecture.
 
 The `viam module upload` command only supports one `platform` argument at a time.
 If you would like to upload your module with support for multiple platforms, you must run a separate `viam module upload` command for each platform.
 Use the _same version number_ when running multiple `upload` commands of the same module code if only the `platform` support differs.
 
+If you specify a platform that includes `any` (such as `any`, `any/amd64`, or `linux/any`), a machine that deploys your module will select the _most restrictive_ architecture from the ones you have provided for your module.
```
Comment (zaporter-work) [inline docs/manage/CLI.md:557]: :+1:

**Review by zaporter-work — APPROVED** [review]
>  but I might suggest presenting a truncated version of each row in the help text output

Good idea. Next time I am doing a PR in that area I'll update it.

Thank you for being so quick with this!

**Review by npentrel — APPROVED** [review]

**Pull request merged**